metadata/src/test/java/org/apache/kafka/controller/QuorumControllerTest.java

```diff
@@ -952,6 +966,7 @@ public void testEarlyControllerResults() throws Throwable {
         }
     }
 
+    @Disabled // TODO: need to fix leader election in LocalLog.
```

Comment:
Did this break from this patch or is it flaky?

Reply:
This PR exposes a bug in `LocalLogManager`, which is that it doesn't really do leader election properly. I started to fix that bug, but then I ended up rewriting big parts of LocalLogManager, which doesn't seem appropriate for 3.3, so I shelved it.